### PR TITLE
DSND-2420: Process filing history status field

### PR DESCRIPTION
**Pull request by sthompsonCH**
## Describe the changes
Implement processing of status field intended for GET response based on YAML rules.

### Related Jira tickets

[DSND-2420](https://companieshouse.atlassian.net/browse/DSND-2420)

### General

- [x] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [x] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [x] Has the code been double-checked against `main`?
- [x] Does the code adhere standards in this repository, including SonarQube checks?

### Testing

- [x] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation

_Where possible, add links to the respective Jira tickets when an item is checked_

- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to deployment repositories?

## Notes to the tester

_Add any testing hints or instructions here._

[DSND-2420]: https://companieshouse.atlassian.net/browse/DSND-2420?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by kinpang-CH — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/filinghistory/api/service/FilingHistoryStatusService.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package uk.gov.companieshouse.filinghistory.api.service;
+
+import java.util.Map;
+import uk.gov.companieshouse.filinghistory.api.model.statusrules.PrefixProperties;
+import uk.gov.companieshouse.filinghistory.api.model.statusrules.StatusRuleProperties;
+import uk.gov.companieshouse.filinghistory.api.service.CompanyNumberStatusProcessor.CompanyNumberAffixes;
+
+public class FilingHistoryStatusService implements StatusService {
```
[the remark this answers is not in the document]
Comment (sthompsonCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/api/service/FilingHistoryStatusService.java:8]: Should this be a component? When I add the annotation it breaks all the tests due to failing to load application context.

The equivalent in delta consumer (`TransformerService`) also does not have a `@Component` annotation.

What do you think?

[the remark this answers is not in the document]
Comment (cgregoryCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/api/service/FilingHistoryStatusService.java:8]: Sorry I forgot we had the bean for the StatusService in config, we could simplify the bean in config to create the StatusRulesProperties and have spring inject it into the FilingHistoryStatusService component.

```@Bean
    public StatusRuleProperties statusRules(@Value("${status.rules}") String rulesFile) throws IOException {
        ObjectMapper mapper = new ObjectMapper(new YAMLFactory());
        InputStream stream = getClass().getResourceAsStream("/%s".formatted(rulesFile));

        return mapper.readValue(stream, StatusRuleProperties.class);
    }```

[the remark this answers is not in the document]
Comment (sthompsonCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/api/service/FilingHistoryStatusService.java:8]: Done

**Comment by ch-code-analysis** [comment]
CI: Security warnings found!

**Review by crowlandsCH — APPROVED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**